credentials/src/main/kotlin/web5/credentials/VerifiableCredential.kt

```diff
@@ -0,0 +1,250 @@
+    // BUT we always include it in our JWKs
+    val publicKey = did.keyManager.getPublicKey(keyAliaz)
+    val algorithm = publicKey.algorithm
+    val jwsAlgorithm = JWSAlgorithm.parse(algorithm.toString())
+
+    val jwtHeader = JWSHeader.Builder(jwsAlgorithm)
+      .keyID(keyAlias)
+      .build()
+
+    val jwtPayload = JWTClaimsSet.Builder()
+      .subject(vcDataModel.credentialSubject.id.toString())
+      .claim("vc", vcDataModel.toMap())
+      .build()
+
+    val jwtObject = SignedJWT(jwtHeader, jwtPayload)
+    val toSign = jwtObject.signingInput
+
+    val signatureBytes = did.keyManager.sign(keyAliaz, toSign)
+
+    val base64UrlEncodedSignature = Base64URL(Convert(signatureBytes).toBase64Url(padding = false))
+    val base64UrlEncodedHeader = jwtHeader.toBase64URL()
+    val base64UrlEncodedPayload = jwtPayload.toPayload().toBase64URL()
+
+    return "$base64UrlEncodedHeader.$base64UrlEncodedPayload.$base64UrlEncodedSignature"
+  }
+
+  /**
+   * Retrieves a field from a verifiable credential by its JSON path.
+   *
+   * @param path The JSON path to the desired field.
+   * @return The field's value if found, or null if the field is not present.
+   */
+  public fun getFieldByJsonPath(path: String): String? {
+    val vcJsonString: String = this.vcDataModel.toJson()
+    return JsonPath.parse(vcJsonString)?.read<String>(path)
+  }
+
+  /**
+   * Converts the current object to its JSON representation.
+   *
+   * @return The JSON representation of the object.
+   */
+  override fun toString(): String {
+    return vcDataModel.toJson()
+  }
+
+  public companion object {
+    private val objectMapper: ObjectMapper = ObjectMapper().apply {
+      registerModule(KotlinModule.Builder().build())
+      setSerializationInclusion(JsonInclude.Include.NON_NULL)
+    }
+
+    /**
+     * Create a [VerifiableCredential] based on the provided parameters.
+     *
+     * @param type The type of the credential, as a [String].
+     * @param issuer The issuer URI of the credential, as a [String].
+     * @param subject The subject URI of the credential, as a [String].
+     * @param data The credential data, as a generic type [T].
+     * @return A [VerifiableCredential] instance.
+     *
+     * Example:
+     * ```
+     * val vc = VerifiableCredential.create("ExampleCredential", "http://example.com/issuers/1", "http://example.com/subjects/1", myData)
+     * ```
+     */
+    public fun <T> create(type: String, issuer: String, subject: String, data: T): VerifiableCredential {
+      val jsonData: JsonNode = objectMapper.valueToTree(data)
+
+      @Suppress("UNCHECKED_CAST")
+      val mapData = objectMapper.treeToValue(jsonData, Map::class.java) as MutableMap<String, Any>
+
+      val credentialSubject = CredentialSubject.builder()
+        .id(URI.create(subject))
+        .claims(mapData)
+        .build()
+
+      val vcDataModel = VcDataModel.builder()
```

Comment:
missing
.issuanceDate(Date())

and optional:
expirationDate
credentialStatus

Also something to think about..
<img width="817" alt="image" src="https://github.com/TBD54566975/web5-kt/assets/5314059/52a48177-a63a-4b55-838d-f2299aa897ca">

the current vc data model package we are using is not compliant with 1.1 because the date format is slightly off.

Reply:
ah good catch! also bummer that danubetech package doesn't enforce this

Reply:
Actually with a second look with Moe it does support 1.1! We all good ✅